drivers/scheduler/k8s/helmScheduler.go

```diff
@@ -0,0 +1,344 @@
+	fileLock := flock.New(strings.Replace(repoFile, filepath.Ext(repoFile), ".lock", 1))
+	lockCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
+		defer fileLock.Unlock()
+	}
+	if err != nil {
+		return err
+	}
+
+	b, err := ioutil.ReadFile(repoFile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	var f repo.File
+	if err := yaml.Unmarshal(b, &f); err != nil {
+		return err
+	}
+
+	if f.Has(name) {
+		logrus.Errorf("repository name (%s) already exists\n", name)
+		return nil
+	}
+
+	c := repo.Entry{
+		Name: name,
+		URL:  url,
+	}
+
+	r, err := repo.NewChartRepository(&c, getter.All(settings))
+	if err != nil {
+		return err
+	}
+
+	if _, err := r.DownloadIndexFile(); err != nil {
+		err := errors.Wrapf(err, "looks like %q is not a valid chart repository or cannot be reached", url)
+		return err
+	}
+
+	f.Update(&c)
+
+	if err := f.WriteFile(repoFile, 0644); err != nil {
+		return err
+	}
+	logrus.Printf("%q has been added to your repositories\n", name)
+	return nil
+}
+
+// RepoUpdate updates charts for all helm repos
+func (k *K8s) RepoUpdate() error {
+	repoFile := settings.RepositoryConfig
+
+	f, err := repo.LoadFile(repoFile)
+	if os.IsNotExist(errors.Cause(err)) || len(f.Repositories) == 0 {
+		return fmt.Errorf("No repositories found, need to add one before updating, err: %v", err)
+	}
+	var repos []*repo.ChartRepository
+	for _, cfg := range f.Repositories {
+		r, err := repo.NewChartRepository(cfg, getter.All(settings))
+		if err != nil {
+			return err
+		}
+		repos = append(repos, r)
+	}
+
+	logrus.Printf("Getting the latest from the chart repositories\n")
+	var wg sync.WaitGroup
+	for _, re := range repos {
+		wg.Add(1)
+		go func(re *repo.ChartRepository) {
+			defer wg.Done()
+			if _, err := re.DownloadIndexFile(); err != nil {
+				logrus.Printf("Unable to get an update from the %q chart repository (%s):\n\t%s\n", re.Config.Name, re.Config.URL, err)
+			} else {
+				logrus.Printf("Successfully got an update from the %q chart repository\n", re.Config.Name)
+			}
+		}(re)
+	}
+	wg.Wait()
+	logrus.Printf("RepoUpdate Completed successfully.\n")
+	return nil
+}
+
+// InstallChart will install the helm chart
+func (k *K8s) InstallChart(repoInfo *scheduler.HelmRepo, args map[string]string) (string, error) {
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(settings.RESTClientGetter(), repoInfo.Namespace, os.Getenv("HELM_DRIVER"), debug); err != nil {
+		return "", err
+	}
+	client := action.NewInstall(actionConfig)
+
+	if client.Version == "" && client.Devel {
+		client.Version = ">0.0.0-0"
+	}
+	client.ReleaseName = repoInfo.ReleaseName
+	cp, err := client.ChartPathOptions.LocateChart(fmt.Sprintf("%s/%s", repoInfo.RepoName, repoInfo.ChartName), settings)
+	if err != nil {
+		return "", err
+	}
+
+	logrus.Printf("chart path: %s\n", cp)
```

Comment:
We can move this to Debug level log or remove it

Reply:
Done